modules/clightning/Makefile

```
@@ -0,0 +1,56 @@
+.PHONY: build-lightning
+
+
+CXX = clang++
+CXXFLAGS += -Wall -Wextra -std=c++20 -fPIC
+
+INCLUDE_DIRS += -I.
+INCLUDE_DIRS += -I../../include
+INCLUDE_DIRS += -I../../external/lightning
+INCLUDE_DIRS += -I../../external/lightning/ccan
+INCLUDE_DIRS += -I../../external/lightning/external/libwally-core/src/secp256k1/include
+
+CXXFLAGS += $(INCLUDE_DIRS)
+
+ROOT_DIR := $(realpath $(dir $(lastword $(MAKEFILE_LIST)))/../..)
+LIGHTNING_DIR := $(ROOT_DIR)/external/lightning
+CLIGHTNING_MAKEFILE := $(ROOT_DIR)/modules/clightning/Makefile-clightning
+CLIGHTNING_LIB_H := $(ROOT_DIR)/modules/clightning/clightning_lib.h
+CLIGHTNING_LIB_C := $(ROOT_DIR)/modules/clightning/clightning_lib.c
+LIBCLIGHTNING_WRAPPER := $(ROOT_DIR)/modules/clightning/libcln.a
+
+all: module.a
+
+$(LIGHTNING_DIR)/config.vars:
+	@echo "Running ./configure in $(LIGHTNING_DIR)..."
+	cd $(LIGHTNING_DIR) && ./configure
+
+
+libcln.a: $(LIGHTNING_DIR)/config.vars
+	@echo "Appending Makefile-clightning to $(LIGHTNING_DIR)/Makefile if not already present..."
+	@marker="$$(head -n 1 $(CLIGHTNING_MAKEFILE))"; \
+	if ! grep -Fxq "$$marker" $(LIGHTNING_DIR)/Makefile; then \
+		echo "Marker not found, appending..."; \
+		cat $(CLIGHTNING_MAKEFILE) >> $(LIGHTNING_DIR)/Makefile; \
+	else \
+		echo "Makefile already includes clightning makefile content."; \
+	fi
+
+	@echo "Building lightning in $(LIGHTNING_DIR)..."
+	cd $(LIGHTNING_DIR) && make libcln.a
+
+	@echo "Copying static library in $(LIGHTNING_DIR)..."
+	cp $(LIGHTNING_DIR)/libcln.a $(LIBCLIGHTNING_WRAPPER)
+
+module.a: module.o libcln.a clightning_lib.o
+	bash ../../merge.sh module.a libcln.a module.o clightning_lib.o
+	ranlib module.a
+
+module.o: module.cpp
```

Also should rebuild if the header file changes:

```suggestion
module.o: module.cpp module.h
```

---

Done